### PR TITLE
Added prefill mode for mamba modules

### PR DESCRIPTION
**Pull request by kpaigwar**
### Problem description
Added prefill mode for seq_len=64 in all the unit tests of mamba modules

### What's changed
1. Shouldn't affect decode graph.
2. Device Perf target got changed for decode as we now pushing residuals on DRAM. Surprisingly, it has improved but not sure if the perf improvements come from rebase.
3. All the prefill tests with seq_len=64 and decode tests are passing with 40 core grid 

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/demos/mamba/tt/model_config.py [hunk 1]
```diff
@@ -4,34 +4,65 @@
 
 import ttnn
 import os
+from models.demos.mamba.reference.args import ModelMode
 
 
-def create_model_config(batch_size, hidden_size):
+def create_model_config(batch_size, hidden_size, mode=ModelMode.DECODE, seq_len=1):
+    # Set the architecture yaml file for 8x8 core_grid
+    # os.environ["WH_ARCH_YAML"] = "wormhole_b0_80_arch_eth_dispatch.yaml"
```
Comment (esmalTT) [inline models/demos/mamba/tt/model_config.py:12]: Remove this comment

File: models/demos/mamba/tests/test_mamba_block.py [hunk 2]
```diff
@@ -29,56 +30,65 @@ def forward(self, x):
...
         (
             "state-spaces/mamba-2.8b",
-            32,
-            0.97,
+            ModelMode.PREFILL,
+            1,
+            64,
+            0.98,
         ),
+        # (
```
Comment (esmalTT) [inline models/demos/mamba/tests/test_mamba_block.py:42]: Uncomment this

File: models/demos/mamba/tt/mamba_block.py [hunk 3]
```diff
@@ -77,6 +76,19 @@ def __init__(self, args: ModelArgs, device, configs, load_fn: Callable):
                 )
             )
 
+        self.use_torch_conv = True
```
Comment (esmalTT) [inline models/demos/mamba/tt/mamba_block.py:79]: Probably can just remove this variable and the condition.

Comment (kpaigwar) [reply models/demos/mamba/tt/mamba_block.py:79]: That was a place holder to select conv version in future.

**Review by esmalTT — APPROVED** [review]

**Pull request merged**